pypesto/objective/history.py

```diff
@@ -1115,10 +1118,10 @@ def _compute_vals_from_trace(self):
         # etc
         max_init_iter = 3
         for it in range(min(len(self.history), max_init_iter)):
-            candidate = self.history.get_fval_trace(it)
+            candidate = self.history.get_fval_trace()[it]
```

Comment:
why is this necessary? passing `it` into the function was designed to allow for efficient access, e.g. not calculating the full trajectory, but only that specific index/indices. Not sure if csv/hdf5 use it, but could potentially.

Reply:
csv uses it, but hdf5 did not that's why I changed it for now into this to be able to use the function in both file formats.

Reply:
oh, I see. That's a bug in the History class: It seems like when `it` was introduced, it was not properly propagated down to hdf5, only csv, therefore the function calls there raise a "Signature does not match base class" warning in pycharm. Could you modify the functions in the Hdf5History?

Reply:
updated